server/authorization.go

```diff
@@ -305,7 +306,32 @@ func addHeaderTransforms(req *authv3.CheckRequest, envRequest *config.Environmen
 				addHeaderValueOption(okResponse, v.Key, v.Value, true)
 			}
 		}
+		if log.DebugEnabled() {
+			log.Debugf(logHeaderValueOptions(okResponse))
+		}
+	}
+}
+
+func logHeaderValueOptions(okResponse *authv3.OkHttpResponse) string {
+	var b strings.Builder
+	b.WriteString("Request header mods:\n")
+	if len(okResponse.Headers) > 0 {
+		for _, h := range okResponse.Headers {
+			addAppend := "="
+			if h.Append.Value {
+				addAppend = "+"
+			}
+			b.WriteString(fmt.Sprintf("  %s %q: %q\n", addAppend, h.Header.Key,
+				golibutil.Truncate(h.Header.Value, config.TruncateDebugRequestValuesAt)))
```

Comment:
Not specific to this PR but I wonder if we should merge two util packages in -golib. WDYT?

Reply:
It's a little awkward, but it doesn't make sense to add envoy / proto imports to -golib.

Reply:
Technically, the util here only imports the generic protobuf types and has nothing specific to Envoy APIs. It's fine to me as is though.

Reply:
Ok, let's just leave it for now.